### ISSUE TITLE
Copy --from doesn't expand argument variable values

### ISSUE THREAD

**Issue by lszinv** [docker/for-linux#895]
* [x] This is a bug report
* [ ] This is a feature request
* [x] I searched existing issues before opening this one

### Expected behavior
When I pass an argument variable into the COPY --from command, it should properly expand the variable into the stored value the same way the FROM command does. 
```
ARG build_image=some_image_name
FROM ${build_image} AS new_image
...
```

### Actual behavior
When I pass an argument variable into the COPY --from command, I receive an error.
```
ARG build_image=some_image_name
...
COPY --from ${build_image} /path/to/file /file/destination
```
output: 
`invalid from flag value ${build_image}: invalid reference format`

### Steps to reproduce the behavior

Sample dockerfile showing the issue:
```
ARG image_name=alpine:20191219

FROM ${image_name}
# COPY --from=${image_name} /usr/lib/libcrypto.so.1.1 /root/libcrypto.so.1.1  # Doesn't work
COPY --from=alpine:20191219 /usr/lib/libcrypto.so.1.1 /root/libcrypto.so.1.1  # Works
```

**Output of `docker version`:**

```
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea838
 Built:             Wed Nov 13 07:29:52 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea838
  Built:            Wed Nov 13 07:28:22 2019
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Output of `docker info`:**

```
Client:
 Debug Mode: false

Server:
 Containers: 20
  Running: 0
  Paused: 0
  Stopped: 20
 Images: 550
 Server Version: 19.03.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: b34a5c8af56e510852c35414db4c1f4fa6172339
 runc version: 3e425f80a8c931f88e6d94a8c831b9d5aa481657
 init version: fec3683
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 4.15.0-1044-aws
 Operating System: Ubuntu 18.04.3 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 15.53GiB
 Name: ip-172-31-0-125
 ID: QYW7:RKAT:72L7:WPNY:JUWJ:D5IX:M6D6:KWGE:IL7U:XIXZ:Q4TA:RSL6
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.)**
Currently running this on an Ubuntu AWS EC2 vm

**Comment by thaJeztah** [comment]
Thanks for reporting; it looks like this is a duplicate of the feature request in https://github.com/moby/moby/issues/34482.

While (to an extend) implementing this would be "easy", there are quite some consequences of making that change, which would prevent the builder from optimising builds (running steps in parallel), as well as causing some ambiguity in the UX. The linked issue provides some more details about why it's not supported, and outlines some of those consequences.

As an alternative, you can create an "alias" stage that uses the build arg. For example:

```dockerfile
ARG BUILD_IMAGE =some_image_name
ARG GO_VERSION=1.14.5

# create aliases for images that can be changed with build-args
FROM ${BUILD_IMAGE} AS build-base
FROM golang:${GO_VERSION} AS my-golang

FROM my-golang AS builder
COPY . . 
RUN go build ./...

FROM scratch
COPY --from=build-base /usr/lib/libcrypto.so.1.1 /root/libcrypto.so.1.1
```


I'm closing this ticket in favour of https://github.com/moby/moby/issues/34482 to prevent the discussion from diverging, but feel free to subscribe to that ticket and to participate in the discussion
